specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json

```diff
@@ -0,0 +1,1214 @@
+          },
+          {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "$ref": "#/parameters/billingProfileIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter by product type. The filter supports 'eq', 'lt', 'gt', 'le', 'ge', and 'and'. It does not currently support 'ne', 'or', or 'not'. Tag filter is a key value pair string where key and value is separated by a colon (:).",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ProductsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/billingAccounts/{billingAccountId}/projects/{projectId}/products": {
+      "get": {
+        "tags": [
+          "Products"
+        ],
+        "x-ms-examples": {
+          "ProductsListByProject": {
+            "$ref": "./examples/ProductsListByProject.json"
+          }
+        },
+        "operationId": "Products_ListByProject",
+        "description": "Lists all products for a user for a billing account and a project",
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "$ref": "#/parameters/projectIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter by product type. The filter supports 'eq', 'lt', 'gt', 'le', 'ge', and 'and'. It does not currently support 'ne', 'or', or 'not'. Tag filter is a key value pair string where key and value is separated by a colon (:).",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ProductsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}/availableBalance": {
+      "get": {
+       "tags": [
+          "Credits"
+        ],
+        "operationId": "AvailableBalanceByBillingProfile_List",
```

Comment:
This looks unlike the other OperationGroup IDs. I'd revise this OperationID to:
`BillingProfiles_ListAvailableBalance` or something similar. That way the operation is available off of the the BillingProfiles client created by AutoRest instead of create a whole new client with a single operation.